### PR TITLE
Search field disable autoSearching, use ENTER keypress to trigger search

### PR DESCRIPTION
**Pull request by kschuste**
### Summary
This fixes issue #952.
Adding the ability to disable the auto-search functionality via a prop and allow for the searching to be triggered via the ENTER keypress.

### Additional Details
Added a new disableAutoSearch prop that is defaulted to false.  Handled onKeyDown event to add the logic for the ENTER keypress.

### CODE REVIEW

File: packages/terra-search-field/tests/nightwatch/search-field-spec.js [hunk 1]
```diff
@@ -74,4 +74,38 @@ module.exports = resizeTo(['tiny', 'small', 'medium', 'large', 'huge', 'enormous
...
+      .expect.element('#search-callback-text').text.to.equal('').after(250);
+    browser
+      .setValue('input[type=search]', 'e')
+      .expect.element('#search-callback-text').text.to.equal('').after(250);
+    browser
+      .setValue('input[type=search]', 'a')
+      .expect.element('#search-callback-text').text.to.equal('').after(250);
+  },
+
+  'Displays the search field and handles ENTER keypress to search ': (browser) => {
```
Comment (emilyrohrbough) [inline packages/terra-search-field/tests/nightwatch/search-field-spec.js]: Could we add a test that searches on Enter with autoSearch enabled?

File: packages/terra-search-field/src/SearchField.jsx [hunk 2]
```diff
@@ -63,11 +74,17 @@ class SearchField extends React.Component {
   handleTextChange(event) {
     this.setState({ searchText: event.target.value });
 
-    if (!this.searchTimeout) {
+    if (!this.searchTimeout && !this.props.disableAutoSearch) {
       this.searchTimeout = setTimeout(this.handleSearch, this.props.searchDelay);
     }
   }
 
+  handleKeyDown(event) {
```
Comment (emilyrohrbough) [inline packages/terra-search-field/src/SearchField.jsx]: I feel like this logic should move into the handleTextChange function.  and the searchTimeout logic should be broken into a resetSearchTimeout function.

Comment (emilyrohrbough) [reply packages/terra-search-field/src/SearchField.jsx]: That being if we keep this,  handleKeyDown should be updated to something a bit more descriptive, like handleSearch

File: packages/terra-search-field/src/SearchField.jsx [hunk 3]
```diff
@@ -63,11 +74,17 @@ class SearchField extends React.Component {
...
+    if (event.nativeEvent.keyCode === KEYCODES.ENTER) {
```
Comment (benbcai) [inline packages/terra-search-field/src/SearchField.jsx]: Is there any difference between event.nativeEvent.keyCode and event.keyCode? I see that some components are using event.nativeEvent.keyCode while others are using event.keyCode

Comment (yuderekyu) [reply packages/terra-search-field/src/SearchField.jsx]: `event.nativeEvent` references the browser's native event, while `event` is a React cross browser synthetic event. Seems that React prefers synthetic events to standardize handling events across browsers.

File: packages/terra-search-field/src/SearchField.jsx (hunk 2 above)
Comment (kschuste) [reply packages/terra-search-field/src/SearchField.jsx]: handleTextChange is only used for when the value inside the search field actually does change, handling the ENTER key does not trigger the onChange event as the value stays the same.  Therefore, added this function to handle the keyPress of ENTER.
We already have a handleSearch function that both handleKeyDown and handleTextChange both call so I think handleKeyDown is descriptive and makes the most sense in that it is handing the down press of the ENTER key

**Comment by PaulSolDev** [comment]
Functional review complete. No issues identified.

**Comment by neilpfeiffer** [comment]
+1 : Marking as Verified, ready for Release.

**Pull request merged**